internal/docs/design.md

```diff
@@ -208,22 +208,44 @@ Concrete types should:
     end users, so that users can't peek into provider-specific error details
     without using `As`. Make sure not to double-wrap.
 
--   Prefix errors with the package name (e.g., `blob: fail`). For most errors,
-    this can be done as part of the wrapping.
-
--   Prefer to keep details of returned `error`s unspecified. The most common
-    case is that the caller will only care whether an operation succeeds or not.
-
--   If certain `error`s are interesting for callers to distinguish, expose
-    predicate functions like
-    [`os.IsNotExist`](https://golang.org/pkg/os/#IsNotExist). This allows the
-    internal representation of the `error` to change over time while being
-    simple to use. In most cases, these predicate functions will translate into
-    a driver interface function to do the check. The concrete type must pass
-    such functions the raw error that the driver returned (not a wrapped error).
-
--   Expose an `ErrorAs` function to allow users to access provider-specific
-    error types.
+- Use `internal/gcerr.New` when wrapping driver errors, like so:
```

Comment:
Nit: can you run `mdformat` on this for consistency before merging?

Reply:
Done.